application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -84,6 +86,32 @@ public TransferQuestionCommand(Config config, ChatGptService chatGptService) {
         this.chatGptService = chatGptService;
     }
 
+
+    public class TitleGenerator {
+
+        // Define regex patterns for leading and trailing quotation marks
+        private static final Pattern LEADING_TRAILING_QUOTES = Pattern.compile("^['\"]|['\"]$");
+
+        public static String generateTitle(String originalMessage, ChatGptService chatGptService) {
+            String chatGptPrompt =
+                    "Summarize the following text into a concise title or heading not more than 4-5 words, remove quotations if any: %s"
+                        .formatted(originalMessage);
+            Optional<String> chatGptResponse = chatGptService.ask(chatGptPrompt, "");
+            String title = chatGptResponse.orElse(createTitle(originalMessage));
+            title = LEADING_TRAILING_QUOTES.matcher(title).replaceAll("");
```

Comment:
Problem is that you're also removing intentional quotations. 

Consider the following example:
`"hello" is being printed but not "hello world"` becomes `hello" is being printed but not "hello world` 

You should probably let ChatGPT handle this entirely and update the prompt to ensure that and test it does it the way we intend it to.

Reply:
In sense, you're really just doing the following on-top of ChatGPT:
```
if(str.startsWith("\"") && str.endsWith("\"")) {
  str.substring(1, str.length - 1);
}
```

Reply:
`"'hello world\""` this is also now just becoming `hello world`

Reply:
We tried to deal at prompt level, unfortunately GPT is very inconsistent and sometimes would end up adding quotations regardless. The best possible way forward is to let GPT do it's thing and then like suraj mentioned look for titles that starting with quotation marks and only remove those that way we handle the edge case.